packages/terra-graphs/src/components/Bar/BarGraph.jsx

```diff
@@ -0,0 +1,59 @@
+/* eslint import/no-unresolved: off */
+import React from 'react';
+import PropTypes from 'prop-types';
+import Carbon from '@cerner/carbon-graphs/lib/js/carbon';
+import '../Graph.module.scss';
+import './BarGraph.module.scss';
+
+const propTypes = {
+  /**
+   * ID for graph created.
+   */
+  graphID: PropTypes.string.isRequired,
+  /**
+   * Graph configuration object
+   */
+  graphConfig: PropTypes.object.isRequired,
+  /**
+   * data to be displayed in graph
+   */
+  dataset: PropTypes.arrayOf(PropTypes.object),
+  /**
+   * timeout to display multiple data contents in specific time interval.
+   */
+  timeout: PropTypes.arrayOf(PropTypes.number),
+};
+
+const BarGraph = ({
+  graphConfig, dataset, graphID, timeout,
+}) => {
+  React.useEffect(() => {
+    const graph = Carbon.api.graph(graphConfig);
+    if (dataset) {
+      if (timeout) {
+        dataset.forEach((data, index) => {
+          setTimeout(
+            () => (graph.graphContainer
+              ? graph.loadContent(Carbon.api.bar(data))
```

Comment:
Getting into edge-case territory here, but we should probably clean up these timeouts if the component unmounts or the effect executes again. Seems like it'd be possible for the timeouts to stomp on each other if the durations were long and the dataset changed quickly.

Something like this maybe:

```jsx
  React.useEffect(() => {
    const graph = Carbon.api.graph(graphConfig);
    const timeoutIds = [];
    
    if (dataset) {
      if (timeout) {
        dataset.forEach((data, index) => {
          const timeoutId = setTimeout(
            () => (graph.graphContainer
              ? graph.loadContent(Carbon.api.bar(data))
              : ''),
            timeout[index],
          );
          
          timeoutIds.push(timeoutId);
        });
      } else {
        dataset.forEach((data) => {
          graph.loadContent(Carbon.api.bar(data));
        });
      }
    }
    
    // the returned cleanup function executes prior subsequent effect 
    // executions (if a config/dataset/timeout were to change) and prior to the unmounting of the component.
    return () => {
      timeoutIds.forEach((id) => { clearTimeout(id); });
    }
  }, [graphConfig, dataset, timeout]);
```